download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -50,4 +50,176 @@ Any backwards incompatible changes are annotated with a [.label.label-danger.lab
 Due to the limited usage of OSGi and the maintenance burden it brings,
 the OptaPlanner jars in the 8.x series no longer include OSGi metadata in their `META-INF/MANIFEST.MF` file.
 
+[.upgrade-recipe-major]
```

Comment:
This is a minor too, very few users use filter classes.

Reply:
Open question: do we consider any of these XML changes to affect the "public API"?
If we do, they need to get the tag `upgrade-recipe-public-api` too, so it becomes `[.upgrade-recipe-minor.upgrade-recipe-public-api]`. I lean towards XML changes do not affect "public API", so they don't need it.

Reply:
On the other hand, what is the recommended way of configuring the solver? Via the `*Config` classes or via the XML?

I would argue, that since we show XML fragments in the documentation and we maintain backward compatibility of the XML format, it's a public API too.